### PR TITLE
Add a to_junit method to standardize the report format of TestSuiteResult

### PR DESCRIPTION
**Pull request by Kranium2002**
## Description

This PR adds to_Junit function to TestSuitResult Class

## Related Issue

This PR is related to issue number #1685 

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [x] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

**Comment by Kranium2002** [comment]
@luca-martial Would you please review my PR ?

File: giskard/core/suite.py [hunk 1]
```diff
@@ -151,6 +152,18 @@ def to_wandb(self, run: Optional["wandb.wandb_sdk.wandb_run.Run"] = None) -> Non
                 "repository for further assistance: https://github.com/Giskard-AI/giskard."
             ) from e
         run.log({"Test suite results/Test-Suite Results": wandb.Table(columns=columns, data=data)})
+      
+    def to_junit(self):
+        """Convert the test suite result to JUnit XML format."""
+        testsuite = Element('testsuite')
+
+        for test in self.results:
+            testcase = SubElement(testsuite, 'testcase', {'name': test.metric_name})
```
Comment (kevinmessiaen) [inline giskard/core/suite.py:161]: The `metric_name` is optional and does not meet the `name` specification:

- name: The label of the rule.

**Review by kevinmessiaen — CHANGES_REQUESTED** [review]
Thanks for the PR, 

my concern is that we lose too much information about the tests when exporting and reading the report don't allow to understand the results.

In my opinion we should generate report like this:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites time="8.605871" tests="3">
    <testsuite name="Test accuracy (model=..., dataset=..., metric=0.5)" time="10.605871">
        <testcase name="metric_name" time="6.605871" />
    </testsuite>
    <testsuite name="Test F1 (model=..., dataset=..., metric=0.5)" time="2">
        <testcase name="metric_name" time="2">
            <failure message="Test failed with metric of 0.3" type="TestFailed">
                <!-- Full Test result information here (metric, output_ds, messages, ...) -->
            </failure>
        </testcase>
    </testsuite>
    <testsuite name="Test with error (model=..., dataset=..., metric=0.5)" time="2">
        <testcase name="metric_name" time="2">
            <failure message="Insert error message here" type="Error">
                <!-- stacktrace given in result.message) -->
            </failure>
        </testcase>
    </testsuite>
</testsuites>
```

File: giskard/core/suite.py (hunk 1 above)
Comment (Kranium2002) [reply giskard/core/suite.py:161]: What do you propose I add instead of metric_name for the name parameter? @kevinmessiaen

Comment (Kranium2002) [reply giskard/core/suite.py:161]: How should I get time for a particular test?

Comment (kevinmessiaen) [reply giskard/core/suite.py:161]: To replace metric name I suggest having something like: `{test_name} ({param_name}: {param_value}, ...)`

Time is not tracked, so either we can keep it empty for now or add a time tracking mechanism in `Suite.run` method.

Comment (Kranium2002) [reply giskard/core/suite.py]: Just changed the to_junit function to have more information.

Sample output:
```<?xml version="1.0" ?>
<testsuites tests="2">
   <testsuite name="Test test1 (metric=0.9)">
      <testcase name="Accuracy" time="0.9">
         <detail name="passed" value="True"/>
         <detail name="props" value="{'prop1': 'value1'}"/>
         <detail name="metric_name" value="Accuracy"/>
         <detail name="metric" value="0.9"/>
         <detail name="missing_count" value="0"/>
         <detail name="missing_percent" value="0"/>
         <detail name="unexpected_count" value="0"/>
         <detail name="unexpected_percent" value="0"/>
         <detail name="unexpected_percent_total" value="0"/>
         <detail name="unexpected_percent_nonmissing" value="0"/>
         <detail name="partial_unexpected_index_list" value="[]"/>
         <detail name="unexpected_index_list" value="[]"/>
         <detail name="number_of_perturbed_rows" value="0"/>
         <detail name="actual_slices_size" value="[]"/>
         <detail name="reference_slices_size" value="[]"/>
         <detail name="output_df" value="   A  B
0  1  4
1  2  5
2  3  6"/>
         <detail name="message" value="Message 1"/>
      </testcase>
   </testsuite>
   <testsuite name="Test test2 (metric=0.8)">
      <testcase name="Precision" time="0.8">
         <failure message="Test failed with metric of 0.8" type="TestFailed">
            <detail name="passed" value="False"/>
            <detail name="props" value="{'prop2': 'value2'}"/>
            <detail name="metric_name" value="Precision"/>
            <detail name="metric" value="0.8"/>
            <detail name="missing_count" value="0"/>
            <detail name="missing_percent" value="0"/>
            <detail name="unexpected_count" value="0"/>
            <detail name="unexpected_percent" value="0"/>
            <detail name="unexpected_percent_total" value="0"/>
            <detail name="unexpected_percent_nonmissing" value="0"/>
            <detail name="partial_unexpected_index_list" value="[]"/>
            <detail name="unexpected_index_list" value="[]"/>
            <detail name="number_of_perturbed_rows" value="0"/>
            <detail name="actual_slices_size" value="[]"/>
            <detail name="reference_slices_size" value="[]"/>
            <detail name="output_df" value="   A  B
0  1  4
1  2  5
2  3  6"/>
            <detail name="message" value="Message 2"/>
         </failure>
      </testcase>
   </testsuite>
</testsuites>```

**Comment by Kranium2002** [comment]
@kevinmessiaen I still need help in implementing the runtime calculation logic. Would you please help me with that?

**Comment by kevinmessiaen** [comment]
> @kevinmessiaen I still need help in implementing the runtime calculation logic. Would you please help me with that?

Sure, let me look at it

**Comment by kevinmessiaen** [comment]
Let's merge without the execution time of each tests at first. I created an issue to tackle the problem of adding execution time: https://github.com/Giskard-AI/giskard/issues/1759

**Review by kevinmessiaen — APPROVED** [review]

**Pull request merged**